xla/stream_executor/integrations/device_mem_allocator.h

```diff
@@ -52,8 +54,10 @@ class DeviceMemAllocator : public tsl::SubAllocator {
     void* ptr = nullptr;
     *bytes_received = num_bytes;
     if (num_bytes > 0) {
-      if (use_unified_memory_) {
+      if (memory_type_ == MemoryType::kUnified) {
```

Comment:
Refactoring unified memory handling could be a separate NFC?

Reply:
This PR doesn't change anything in the way unified memory is handled. The change here is to the interface of `DeviceMemAllocator`. Previously, the allocator used a bool to determine whether to use one of two options: regular cuda memory or unified memory. This PR is adding a third option to use `ncclMemAlloc` memory, so I changed the bool to an enum.